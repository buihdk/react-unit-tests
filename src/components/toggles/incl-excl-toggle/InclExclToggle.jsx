import React, { memo } from 'react';
import I18n from 'i18n-js';
import PropTypes from 'prop-types';
import { Radio } from 'antd';

const InclExclToggle = ({ form: { getFieldDecorator }, type }) => (
  <div>
    {getFieldDecorator(`${type}InclExclToggle`, { initialValue: 'include' })(
      <Radio.Group>
        <Radio value="include">{I18n.t('types.siteZoneType.include')}</Radio>
        <Radio value="exclude">{I18n.t('types.siteZoneType.exclude')}</Radio>
      </Radio.Group>,
    )}
  </div>
);

InclExclToggle.propTypes = {
  form: PropTypes.shape({
    getFieldDecorator: PropTypes.func.isRequired,
  }).isRequired,
  type: PropTypes.string.isRequired,
};

const MemoInclExclToggle = memo(InclExclToggle);
MemoInclExclToggle.displayName = 'InclExclToggle';

export default MemoInclExclToggle;
