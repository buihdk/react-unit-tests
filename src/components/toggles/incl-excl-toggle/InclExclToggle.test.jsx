import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import InclExclToggle from './InclExclToggle';

const props = {
  form: {
    getFieldDecorator: () => element => element,
  },
  type: 'segment',
};

describe('InclExclToggle', () => {
  test('renders without crashing ', () => {
    const wrapper = shallow(<InclExclToggle {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
