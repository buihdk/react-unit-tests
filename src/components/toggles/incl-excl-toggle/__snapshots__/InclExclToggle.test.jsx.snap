// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InclExclToggle renders without crashing  1`] = `
<div>
  <Memo([object Object])>
    <Radio
      type="radio"
      value="include"
    >
      [missing "en.types.siteZoneType.include" translation]
    </Radio>
    <Radio
      type="radio"
      value="exclude"
    >
      [missing "en.types.siteZoneType.exclude" translation]
    </Radio>
  </Memo([object Object])>
</div>
`;
